Content.Server/Anomaly/AnomalySynchronizerSystem.cs

```diff
@@ -0,0 +1,158 @@
+using Content.Server.Anomaly.Components;
+using Content.Server.DeviceLinking.Systems;
+using Content.Server.Power.Components;
+using Content.Shared.Anomaly.Components;
+using Content.Shared.Interaction;
+using Content.Shared.Popups;
+
+namespace Content.Server.Anomaly;
+
+/// <summary>
+/// a device that allows you to translate anomaly activity into multitool signals.
+/// </summary>
+public sealed partial class AnomalySynchronizerSystem : EntitySystem
+{
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly EntityLookupSystem _entityLookup = default!;
+    [Dependency] private readonly DeviceLinkSystem _signalSystem = default!;
+    [Dependency] private readonly AnomalySystem _anomaly = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<AnomalySynchronizerComponent, InteractHandEvent>(OnInteractHand);
+        SubscribeLocalEvent<AnomalySynchronizerComponent, PowerChangedEvent>(OnPowerChanged);
+
+        SubscribeLocalEvent<AnomalyPulseEvent>(OnAnomalyPulse);
+        SubscribeLocalEvent<AnomalySupercriticalEvent>(OnAnomalySupercritical);
+        SubscribeLocalEvent<AnomalyStabilityChangedEvent>(OnAnomalyStabilityChanged);
+    }
+
+    private void OnPowerChanged(EntityUid uid, AnomalySynchronizerComponent component, ref PowerChangedEvent args)
+    {
+        if (args.Powered)
+            return;
+
+        if (!TryComp<AnomalyComponent>(component.ConnectedAnomaly, out var anomaly))
+            return;
+
+        _anomaly.DoAnomalyPulse(component.ConnectedAnomaly.Value, anomaly);
+        DisconnetFromAnomaly(uid, component, anomaly);
+    }
+
+    private void OnInteractHand(EntityUid uid, AnomalySynchronizerComponent component, InteractHandEvent args)
+    {
+        foreach (var entity in _entityLookup.GetEntitiesInRange(uid, 0.15f)) //is the radius of one tile. It must not be set higher, otherwise the anomaly can be moved from tile to tile
+        {
+            if (TryComp<AnomalyComponent>(entity, out var anomaly))
+            {
+                ConnectToAnomaly(uid, component, anomaly);
```

Comment:
pass in entity

Reply:
wat

Reply:
ConnectToAnomaly(uid, component, entity, anomaly);